spec/presenters/activity_presenter_spec.rb

```diff
@@ -3,7 +3,29 @@
 require "rails_helper"
 
 RSpec.describe ActivityPresenter do
+  RSpec.shared_examples "a code translator" do |field, args, field_type|
+    let(:field) { field }
+    let(:args) { args.reverse_merge(source: "iati") }
+    let(:cast) do
```

Comment:
Thanks Stu for explaining the layers of meta-programming! We agreed to replace this part with a less exciting but easier to follow `if` statement.